### PR TITLE
Social Image Update: Absolute Positioning, Scale Calculation, and Pipeline Connectors

### PR DESCRIPTION
**Pull request by roshan-gh**
This pull request introduces significant changes to the social image rendering.

**1- Absolute Positioning:** The entire CSS styling for both nodes and edges has been transitioned to use absolute positioning. This change aims to enhance the predictability and precision of the layout.

**2- Scale Calculation from Scratch:** The scale calculation logic has been re-implemented from scratch. This ensures a more accurate representation of elements on the social image, taking into account absolute positioning and other layout changes.

**3- Pipeline Connectors:** A new feature has been introduced to draw connectors between pipelines to align the image with the client side visualization.

These modifications collectively improve the overall rendering of social images, offering a more stable and visually appealing experience. Testing and feedback on the updated version are appreciated.

### CODE REVIEW

File: packages/otelbin/src/app/og/ParentsNode.tsx [hunk 1]
```diff
@@ -29,61 +25,45 @@ const ParentsNode = ({ nodeData, nodes }: { nodeData: Node; nodes?: Node[] }) =>
...
+		const edgesToDraw: IEdge[] = [];
+
+		if (!parentNode) {
+			return;
+		}
 
-	function calcSVGPath(side: string, nodes?: Node[]) {
-		const nodesCount = nodes?.length ?? 0;
-		const height = nodesCount * (nodeHeight + nodeTotalMargin) - 40;
+		edges.map((edge) => {
```
Comment (bripkens) [inline packages/otelbin/src/app/og/ParentsNode.tsx:46]: **Nitpick:** Looks like we are ignoring the `map(…)` result. Use `forEach(…)` instead?

Actually, no, instead I'd suggest to return a value in the `map(…)` step and to remove `edgesToDraw`

File: packages/otelbin/src/app/og/ParentsNode.tsx [hunk 2]
```diff
@@ -29,61 +25,45 @@ const ParentsNode = ({ nodeData, nodes }: { nodeData: Node; nodes?: Node[] }) =>
...
-		return nodesCount * (nodeHeight + nodeTotalMargin) - nodeTotalMargin;
-	}
+	function drawSvgInsideParentNode(edges: Edge[], parentNode: Node) {
+		const padding = 10;
+		const nodeWidth = 120 + padding;
+		const halfNodeHeight = 80 / 2;
+		const edgesToDraw: IEdge[] = [];
+
+		if (!parentNode) {
+			return;
```
Comment (bripkens) [inline packages/otelbin/src/app/og/ParentsNode.tsx:43]: **Issue:** The function has an inconsistent return type. Return an empty array here?

File: packages/otelbin/src/app/s/[id]/img/route.tsx [hunk 3]
```diff
@@ -32,8 +34,57 @@ export async function GET(request: NextRequest) {
...
+				if (sourceNode && targetNode && sourceParent && targetParent) {
+					const sourcePosition = {
+						x: sourceParentPosition.x + sourceNode.position.x + nodeWidth + padding + totalXOffset,
+						y: sourceParentPosition.y + sourceNode.position.y + halfNodeHeight,
+					};
+					const targetPosition = {
+						x: targetParentPosition.x + targetNode?.position.x - padding + totalXOffset,
+						y: targetParentPosition.y + targetNode?.position.y + halfNodeHeight,
+					};
+					edgesToDraw.push({ edge: edge, sourcePosition: sourcePosition, targetPosition: targetPosition });
```
Comment (bripkens) [inline packages/otelbin/src/app/s/[id]/img/route.tsx:80]: **Issue:** Similar code path to the one above, where the intention isn't clear.

File: packages/otelbin/src/app/s/[id]/metadataUtils.test.ts [hunk 4]
```diff
@@ -45,80 +44,6 @@ describe("extractComponents", () => {
 	});
 });
 
-describe("calcScale", () => {
```
Comment (bripkens) [inline packages/otelbin/src/app/s/[id]/metadataUtils.test.ts:48]: **Question:** This test got removed, but the function continues to exist. Why remove it?

**Review by bripkens — COMMENTED** [review]
**Praise:** It works better now and the line styles are aligned! Nice :)

The feedback is basically just about code style/quality. Also see Sonar's feedback in that regard.

File: packages/otelbin/src/app/s/[id]/metadataUtils.test.ts (hunk 4 above)
Comment (roshan-gh) [reply packages/otelbin/src/app/s/[id]/metadataUtils.test.ts:48]: The entire logic for calculating the scale has been revised. New tests will be added.

**Comment by roshan-gh** [comment]
@bripkens Addressed and resolved comments; included tests for related functions.

**Review by bripkens — COMMENTED** [review]
LGTM aside from the Sonar issues. Some of the code got quite complex. Let's fix that before merge.

**Review by bripkens — APPROVED** [review]
LGTM, lets get this in! :)

**Pull request merged**